book/src/read-only-read-write-accounts.md

```diff
@@ -0,0 +1,136 @@
+# Read-Only/Read-Write Transaction Accounts
+
+This design covers handling of read-only and read-write accounts in the
+[runtime](runtime.md).  Accounts already distinguish themselves as read-only or
+read-write based on the program id specified by the transaction's instruction.
+Programs must treat accounts that are not owned by them as read-only.
+
+To identify read-only accounts by program id would require the Account to be
+fetched and loaded from disk.  This operation is expensive and while it is
+occurring the runtime would have to reject any transactions referencing the same
+account.
+
+The proposal introduces a `read_only_accounts` field to the Transaction
+structure and removes the `program_ids` dedicated vector for program accounts.
+
+This design doesn't change the runtime transaction processing rules.
+Programs still can't write or spend accounts that they do not own.  But it
+allows the runtime to optimistically take the right lock for each account
+specified in the transaction before loading the accounts from storage.
+
+Accounts selected as read-write by the transaction can still be treated as
```

Comment:
The second sentence in this paragraph appears to be saying the same thing as the first sentence.  De-dup?